### DISCUSSION TITLE
Would it be possible to run ssr in v8go (not node)?

### DISCUSSION THREAD

**Discussion by JanGordon** [lit/lit#4193] [Q&A]
I would like to run my ssr script in v8go but the ssr package uses node built in packages which seems unnecessary for ssr. For example, I don't really understand why node:fs is needed for ssr.

**Comment by augustjk** [comment]
Depends on what parts of SSR you need. `node:fs` is only used as part of the module loader when opting for isolated rendering in Node's VM module.

I think we do require `node:buffer` for base64 encoding though. That, or a different way to do the conversion will need to be supplied.

Does v8go provide any global API at all? There have been calls for supporting SSR in other environments like service workers, cloud/edge workers, other runtimes like Deno or Bun. It would be nice if we could have a build that we know works least with these common APIs https://common-min-api.proposal.wintercg.org/.

**Reply by JanGordon** [reply]
Yes, v8go does have a global API, and so far I've implemented a buffer polyfill for it (not complete, but I think it covers all lit's needs, just base64 encoding). 

node:FS can be avoided but there are other external requires when I bundle lit with esbuild. For example, there is node:fetch, so I assume I would have to make or find a polyfill for that and then all would work?

Although this is possible, I just think it would be easier  if lit used more engine generalised libraries instead of node specific ones. 

And that link looks promising, haven't heard of that before, but I wish that everyone used complied to that.
